### DISCUSSION TITLE
{ nullable: 'itemsAndList' } throws Error using null value in list

### DISCUSSION THREAD

**Discussion by karladler** [MichalLytek/type-graphql#1324] [Q&A]
Help needed: I have my resolver defined as following:

```typescript
@Query(() => [Company])
  async allCompanies(
    @Ctx() ctx: Context,
      @Arg('storeTypes', () => [String], { nullable: 'itemsAndList' }) storeTypes?: Array<string | null>,
  ): Promise<Company[]> { 
  // ...
```

 but when I use `null` in my query I get an `Argument Validation Error` 

```gql
query allCompanies {
  allCompanies(storeTypes:["otherBusiness", null]) {
    uuid 
    name
  }
}
```

- Works without using `null`
- Works without storeTypes param

Edit: setting {validate: false} in arg options resolves the problem, but it' still weird behaviour since validation usually goes along with the typescript settings for the arguments. I assume this is a bug.

**Comment by MichalLytek** [comment]
> Argument Validation Error

Can you disable `validate: false`? Maybe `class-validator` is trying to do something with args and fails on null.

**Reply by karladler** [reply]
yeah, as written in the "edit" it works now with `{validate: false}`. I couldn't find in the code where class validator jumps in, but there seems to be still some validation even with validate:false since it would complain e.g. when using integers instead of strings, which I'm glad that it's there, but I don't get the fully logic here ...

**Reply by MichalLytek** [reply]
"when using integers instead of strings"
Then the GraphQL type validation would kick in I think
